iothub_client/samples/iothub_convenience_sample/iothub_convenience_sample.c

```diff
@@ -129,13 +174,35 @@ static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void
     (void)printf("Confirmation callback received for message %zu with result %s\r\n", g_message_count_send_confirmations, ENUM_TO_STRING(IOTHUB_CLIENT_CONFIRMATION_RESULT, result));
 }
 
+
+static int check_for_input_func(void *arg)
+{
+	arg;
+	getchar();
```

Comment:
We should avoid getchar() because an embedded device that doesn't have an input mechanism this won't work on.

Reply:
Without a input mechanism, I'm not sure how we would want this simulation to work. Some of the other samples (Node.js, C#, Python) are just using an infinite loop instructing the customer to use Ctrl+C to terminate. Strangely, the code does still include input mechanism.  

Would you recommend just using an infinite loop with Ctrl+C to terminate?

Reply:
@jspaith I went ahead and removed the background thread. Let me know if you disagree with using an infinite loop here. Thanks